src/main/java/cz/tstrecha/timetracker/service/impl/UserServiceImpl.java

```diff
@@ -171,4 +172,24 @@ public LoginResponseDTO loginUser(LoginRequestDTO loginRequest) {
         var refreshToken = authenticationService.generateRefreshToken(user.getId(), user.getId());
         return new LoginResponseDTO(true, token, refreshToken);
     }
+
+    @Override
+    @Transactional
+    public LoginResponseDTO changePassword(PasswordChangeDTO passwordChangeDTO, UserContext userContext) {
+        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(userContext.getEmail(), passwordChangeDTO.getPassword()));
+        var user = userRepository.findByEmail(userContext.getEmail())
+                .orElseThrow(() -> new UsernameNotFoundException("No user exists for email [" + userContext.getEmail() + "]"));
+        if (IntStream.of(0, passwordChangeDTO.getNewPassword().length() - 1).noneMatch(i -> Character.isDigit(passwordChangeDTO.getNewPassword().charAt(i)))) {
```

Comment:
DRY => This code already exists when registering -> move to separate method validatePassword()